neonvm/controllers/vm_controller.go

```diff
@@ -812,6 +861,23 @@ func (r *VMReconciler) doReconcile(ctx context.Context, vm *vmv1.VirtualMachine)
 	return nil
 }
 
+func pickMemoryProvider(config *ReconcilerConfig, vm *vmv1.VirtualMachine) vmv1.MemoryProvider {
+	if p := vm.Spec.Guest.MemoryProvider; p != nil {
+		return *p
+	}
+	if p := vm.Status.MemoryProvider; p != nil {
```

Comment:
Won't this prevent us from changing from one provider to the other by deleting the pod, because the old value be preserved through Status?

Reply:
the tl;dr on that is that we reset `.status.memoryProvider` in the same place we reset `.status.podName` (there's some method on `VirtualMachineStatus` that does it for us).

I tested it locally and it works as expected :)